src/Search.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useRef } from 'react';
+import './Components/Searchbar/Searchbar.scss'
+import search from './assets/search.svg'
+
+// can do as type or interface -- either way works
+type SearchProps = {
+    findPostsWithSearchTerm: (searchTerm: string) => void;
+}
+// interface SearchProps {
+//     findPostsWithSearchTerm: (searchTerm: string) => void
+// }
+
+const Search: React.FC<SearchProps> = (props) => {
+    // refers to the type of data that will be stored inside the ref, default or initialized with null
+    const searchInputRef = useRef<HTMLInputElement>(null);
+
+
+    // handleChange = (event: { target: { name: string, value: string } ; }) => {
+    //   this.setState({ searchInput: event.target.value })
+    // }
+    //
+    const handleSubmit = (event: React.FormEvent) => {
+      event.preventDefault();
+      const enteredText = searchInputRef.current!.value; // ! tells TS to chill until it's set (it's okay to be null)
+      console.log("ENTERED TEXT: ", enteredText)
+        props.findPostsWithSearchTerm(enteredText)
+    }
+
+    return (
+        <form className='searchbar'>
+            <input
+                data-cy='searchbar-input'
+                className='search-input'
+                type='text'
+                placeholder='Body of your post*'
+                name='content'
+                ref={searchInputRef}
+                // value={this.state.searchInput}
+                // onChange={this.handleChange}
+            />
+            <button className='search-button' onClick={handleSubmit}>
+                <img className='search-image' alt='search-icon' src={search}/>
+            </button>
+        </form>
+    )
+}
+
```

Comment:
This is a different way to implement TS, using a functional component to pass the info up to App, instead of a class component. I'm learning TS as I go, so I thought I'd use a different implementation than what you've used.

Reply:
I'd love to talk through this way of doing it. I'm not super familiar with useRef but interested to learn more. They TypeScript makes sense though

Reply:
To be honest, I'm not super familiar with it either. It's a React hook. I'm actually curious to talk about how TS implements a component here (`React.FC` instead of the way you've done it).